sound/soc/sof/topology.c

```diff
@@ -2651,11 +2651,5 @@ void snd_sof_free_topology(struct snd_sof_dev *sdev)
 		kfree(sroute->private);
 		kfree(sroute);
 	}
-
-	ret = snd_soc_tplg_component_remove(sdev->component,
-					    SND_SOC_TPLG_INDEX_ALL);
-	if (ret < 0)
-		dev_err(sdev->dev,
-			"error: tplg component free failed %d\n", ret);
```

Comment:
I have mixed feelings about this.
Is the framework design intended to release all resources (similar to devm_ for memory allocation) or does it call tplg_component_remove for extra safety?